Source/Library/ApplicationController.cpp

```diff
@@ -0,0 +1,98 @@
+#include "TicTacToe/Library/ApplicationController.hpp"
+#include "TicTacToe/Library/DrawConditionWindow.hpp"
+#include "TicTacToe/Library/TitleWindow.hpp"
+#include "TicTacToe/Library/WinConditionWindow.hpp"
+
+using namespace TicTacToe;
+
+ApplicationController::ApplicationController()
+  : m_x_badge_count(0),
+    m_o_badge_count(0),
+    m_application_window() {
+  m_application_window.setWindowFlags(Qt::MSWindowsFixedSizeDialogHint);
+  auto layout = new QVBoxLayout(&m_application_window);
+  layout->setContentsMargins(0, 0, 0, 0);
+  layout->setSpacing(0);
+  m_application_window.setLayout(layout);
+  m_application_window.show();
+  show_title_window();
+}
+
+ApplicationController::~ApplicationController() {
+  auto child = m_application_window.layout()->itemAt(0)->widget();
```

Comment:
The game board window belongs to m_game_board_controller, and it also is a child window of m_application_window, so it can be managed by these two objects at the same time. If the destructor is removed and the application quits while the game board window is showing, m_game_board_controller is released along with the game board window being destroyed and m_application_window also tries to destroy its child windows, so the game board window will be deleted two times. What the destructor does is to disconnect the game board window from the application window in order to prevent the game board window from being destroyed a second time.

Reply:
Okay, makes sense. As a first attempt to fix the issue, declare ```m_game_board_controller``` before ```m_application_window``` so that ```m_application_window``` is destroyed first. If that doesn't fix the problem then you can keep this destructor the way it is.

Reply:
It's a good idea and it works. It's my fault. Actually m_game_board_controller is destroyed first when it is declared after m_application_window.